purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -436,6 +429,7 @@ class Purchases internal constructor(
         startPurchase(activity, storeProduct, purchaseOption, null, callback)
     }
 
+    // TODOBC5: remove storeProduct parameter
```

Comment:
I think we could just duplicate the information we need to purchase/send to the backend in the `GooglePurchaseOption`. I believe that would be the original `productDetails` + `productId`. That way we would avoid any circular references, but it might require some refactor to do this.